// @medipass/web-sdk/lib/utils/base-urls/core.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var coreBaseUrls = {
  au: {
    // @ts-expect-error TS(2591): Cannot find name 'process'. Do you need to install... Remove this comment to see the full error message
    local: "http://localhost:" + (process.env.REACT_APP_LOCAL_PORT || 2673) + "/",
    dev: 'https://dev-api-au.medipass.io/',
    'pre-prod': 'https://api-au-blue.medipass.io/',
    prod: 'https://api-au.medipass.io/',
    staging: 'https://stg-api-au.medipass.io/',
    perf: 'https://perf-api-au.medipass.io/'
  },
  uk: {
    // @ts-expect-error TS(2591): Cannot find name 'process'. Do you need to install... Remove this comment to see the full error message
    local: "http://localhost:" + (process.env.REACT_APP_LOCAL_PORT || 2673) + "/",
    dev: 'https://api.dev.uk.medipass.io/',
    'pre-prod': 'https://api.blue.uk.medipass.io/',
    prod: 'https://api.uk.medipass.io/',
    staging: 'https://api.stg.uk.medipass.io/',
    perf: 'https://api.perf.uk.medipass.io/'
  }
};

exports.default = coreBaseUrls;
