// @medipass/web-sdk/lib/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('@babel/runtime/helpers/defineProperty');
require('@babel/runtime/regenerator');
require('@babel/runtime/helpers/asyncToGenerator');
require('lodash/merge');
require('lodash/uniq');
require('axios');
require('date-fns/differenceInSeconds');
require('jwt-decode');
require('lodash/forOwn');
require('lodash/get');
require('@medipass/utils/sentry');
require('./utils/get-axios-error.js');
var request = require('./staff-28f66bde.js');
require('@babel/runtime/helpers/inheritsLoose');
require('@babel/runtime/helpers/wrapNativeSuper');
require('./utils/param-error.js');
require('./utils/does-param-exist.js');
require('lodash/omitBy');
require('lodash/isEmpty');
require('store/src/store-engine');
require('store/storages/localStorage');
require('store/storages/sessionStorage');
require('masked');
require('@medipass/utils/sensitive-keys');
require('./services/storage.js');
require('platform');
require('@medipass/utils/constants');
require('braintree-web/client');
require('braintree-web/google-payment');
require('braintree-web/apple-pay');
require('@medipass/pify');
require('@medipass/utils/get-env');
var filterOption = require('./types/filter-option.js');
require('@medipass/utils/funders');
require('compare-versions');
require('./utils/base-urls/core.js');
require('./utils/base-urls/risk.js');
require('./utils/base-urls/index.js');
require('lodash/flatten');
require('js-cookie');
var regions = require('./regions-0da90c43.js');



exports.API_VER = request.API_VER;
exports.API_VERSIONS = request.API_VERSIONS;
exports.APP_ID = request.APP_ID;
exports.CONTENT_TYPE = request.CONTENT_TYPE;
exports.DEFAULT_ENV = request.DEFAULT_ENV;
exports.DEFAULT_REGION = request.DEFAULT_REGION;
exports.DEFAULT_VERSION = request.DEFAULT_VERSION;
exports.HEADER__API_VER = request.HEADER__API_VER;
exports.HEADER__APP_ID = request.HEADER__APP_ID;
exports.HEADER__APP_VERSION = request.HEADER__APP_VERSION;
exports.HEADER__DEVICE_ID = request.HEADER__DEVICE_ID;
exports.HEADER__INTERCOM_HASH = request.HEADER__INTERCOM_HASH;
exports.HEADER__SDK_VERSION = request.HEADER__SDK_VERSION;
exports.HEADER__TOKEN = request.HEADER__TOKEN;
exports._isTokenExpired = request._isTokenExpired;
exports.abr = request.abr;
exports.accounts = request.accounts;
exports.actions = request.actions;
exports.apiKeys = request.apiKeys;
exports.applicationUtils = request.applicationUtils;
exports.applications = request.applications;
exports.attachments = request.attachments;
exports.auth = request.auth;
exports.aws = request.aws;
exports.bulkSubmits = request.bulkSubmits;
exports.bulkUploads = request.bulkUploads;
exports.businessProfiles = request.businessProfiles;
exports.businesses = request.businesses;
exports.claimItems = request.claimItems;
exports.default = request.sdk;
exports.devices = request.devices;
exports.docAccounts = request.docAccounts;
exports.documents = request.documents;
exports.emails = request.emails;
exports.filterOptions = request.filterOptions;
exports.formApplicationTemplates = request.formApplicationTemplates;
exports.formApplications = request.formApplications;
exports.forms = request.forms;
exports.funderServices = request.funderServices;
exports.funderSettings = request.funderSettings;
exports.funderValues = request.funderValues;
exports.funders = request.funders;
exports.futures = request.futures;
exports.healthFundAccounts = request.healthFundAccounts;
exports.healthFundSettings = request.healthFundSettings;
exports.healthfunds = request.healthfunds;
exports.icdItems = request.icdItems;
exports.invoiceScans = request.invoiceScans;
exports.jhcs = request.jhcs;
exports.kyc = request.kyc;
exports.logs = request.logs;
exports.members = request.members;
exports.messageMappings = request.messageMappings;
exports.minions = request.minions;
exports.modalities = request.modalities;
exports.notes = request.notes;
exports.onboardingApplications = request.onboardingApplications;
exports.onboardingRequests = request.onboardingRequests;
exports.ops = request.ops;
exports.organisations = request.organisations;
exports.partners = request.partners;
exports.patients = request.patients;
exports.payments = request.payments;
exports.pms = request.pms;
exports.practiceProfiles = request.practiceProfiles;
exports.practiceTypes = request.practiceTypes;
exports.practices = request.practices;
exports.products = request.products;
exports.professionalCategories = request.professionalCategories;
exports.professionsUtils = request.professionsUtils;
exports.providerNumberTypes = request.providerNumberTypes;
exports.providerRegistrationTypes = request.providerRegistrationTypes;
exports.providerRequests = request.providerRequests;
exports.providerTypes = request.providerTypes;
exports.refSources = request.refSources;
exports.risk = request.risk;
exports.rnas = request.rnas;
exports.roles = request.roles;
exports.sdkClient = request.sdkClient;
exports.selfCheckoutSessions = request.selfCheckoutSessions;
exports.services = request.services;
exports.settlements = request.settlements;
exports.signUp = request.signUp$1;
exports.specialties = request.specialties;
exports.staff = request.staff;
exports.staffProfiles = request.staffProfiles;
exports.staffUtils = request.staffUtils;
exports.statistics = request.statistics;
exports.subscriptions = request.subscriptions;
exports.terminals = request.terminals;
exports.transactionExport = request.transactionExport;
exports.transactions = request.transactions;
exports.triggers = request.triggers;
exports.vendors = request.vendors;
exports.verify = request.verify;
exports.versions = request.versions;
exports.workers = request.workers;
exports.workflowExceptions = request.workflowExceptions;
Object.defineProperty(exports, 'FilterGroup', {
	enumerable: true,
	get: function () {
		return filterOption.FilterGroup;
	}
});
exports.regions = regions.regions;
